### PR TITLE
Add Input.t generation utilities

### PR DESCRIPTION
**Pull request by ihodes**
New executable biokepi-input-generator which takes a host and a dir of
fastq.gz and generates JSON serialization of Input.t for input into
Pipelines.ts.

Library supporting this at Biokepi.EDSL.Library.Input.Derive

---
This change is [Reviewable](https://reviewable.io/reviews/hammerlab/biokepi/311)

### CODE REVIEW

**Comment by ihodes** [comment]
@arahuja this might be useful for you

File: src/app/input_gen.ml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+open Biokepi.EDSL.Library.Input
+open Nonstd
+module String = Sosa.Native_string
+
+let cmd () =
+  let open Cmdliner in
+  let version = "0.0.0" in
+  let doc = "Submit a PGV Pipeline" in
```
Comment (smondet) [inline src/app/input_gen.ml]: Could take the version from `Biokepi_run_environment.Metadata.version`?

The `doc` argument is wrong :)

File: src/app/input_gen.ml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  let man = [
+    `S "Description";
+    `P "The PGV Pipeline takes FASTQs and runs them through the PGV pipeline.";
```
Comment (smondet) [inline src/app/input_gen.ml]: This one too :)

File: src/pipeline_edsl/pipeline_library.mli [hunk 3]
```diff
@@ -34,6 +33,24 @@ module Input : sig
   val to_yojson : t -> Yojson.Safe.json
   val of_yojson : Yojson.Safe.json -> (t, string) Pvem.Result.t
 
+  module Derive : sig
```
Comment (smondet) [inline src/pipeline_edsl/pipeline_library.mli]: Could you move the doc-comments from the `ml` to the `mli`?

File: OMakeroot [hunk 4]
```diff
@@ -88,7 +89,8 @@ install: install-lib-biokepi_run_environment \
          install-lib-biokepi_bfx_tools \
          install-lib-biokepi_pipeline_edsl \
          install-lib-biokepi \
-         install-app-biokepi-demo
+         install-app-biokepi-demo \
+         install-app-biokepi-input-generator
```
Comment (smondet) [inline OMakeroot]: we need the `uninstall-app-biokepi-input-generator` also a few lines below

File: src/app/input_gen.ml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    let doc = "Directory on `host` where tumor fastq.gzs can be found." in
+    Arg.(required & opt (some string) None & info ["directory"; "d"] ~doc)
+  in
+  Term.(pure (fun name host directory ->
+      let host = match Ketrew_pure.Host.of_string host with
+      | `Ok host -> host
+      | `Error msg -> printf "Error parsing host.\n%!"; exit 1
+      in
+      let open Pvem_lwt_unix.Deferred_result in
+      Derive.fastqs ~host:host directory
```
Comment (smondet) [inline src/app/input_gen.ml]: Having the option to pass false to `~paired_end` from the CLI would be nice also

**Comment by ihodes** [comment]
Done, back to you

**Pull request merged**